v2/tools/generator/internal/codegen/pipeline/add_kubernetes_exporter.go

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/functions"
+)
+
+const AddKubernetesExporterStageID = "addKubernetesExporter"
+
+func AddKubernetesExporter(idFactory astmodel.IdentifierFactory) *Stage {
+	return NewStage(
+		AddKubernetesExporterStageID,
+		"Adds the KubernetesExporter interface to resources that need it",
+		func(ctx context.Context, state *State) (*State, error) {
+			defs := state.Definitions()
+			updateDefs := make(astmodel.TypeDefinitionSet)
+
+			mappings := state.GeneratedConfigMaps()
+
+			for _, def := range astmodel.FindResourceDefinitions(defs) {
+				resourceType := def.Type().(*astmodel.ResourceType)
```

Comment:
While I think this works now, you should probably use `astmodel.AsResourceType()` in case it's wrapped with something else by a later change to the pipeline.

Reply:
Done